commons/protocol/src/main/scala/sbt/protocol/Protocol.scala

```diff
@@ -139,6 +139,16 @@ final case class ClientInfo(uuid: String, configName: String, humanReadableName:
 final case class RegisterClientRequest(info: ClientInfo) extends Request
 final case class RegisterClientResponse(info: ServerInfo) extends Response
 
+/**
+ * If there's a request or event we don't understand because it's from
+ * a newer version of the protocol, then we put it here. Unknown responses
+ * are converted to ErrorResponse and generally aren't "allowed" because
+ * it's not backward compatible to change the possible responses to an
+ * existing request, so this will be a request or event.
+ * This message is never sent over the wire.
+ */
+final case class UnknownMessage(serialized: SerializedValue) extends Message
```

Comment:
for some reason this looks a lot less ugly than it used to, prior to sbt-serialization.